### ISSUE TITLE
linker doesn't work as function argument in monitor

### ISSUE THREAD

**Issue by erik-roberts** [DynaSim/DynaSim#490]
eg
```
monitor fn1(@linker1,@linker2)
monitor fn2(var1,@linker1,@linker2)
```

**Comment by erik-roberts** [comment]
check dsGenerateModel, dsParseModelEquations, dsPropagateNamespaces

**Comment by kupiqu** [comment]
I think `monitor monitorTestFnLabel = testFn(@linker1,@linker2...)` is a pretty good standard in this case, but this should be documented

**Comment by kupiqu** [comment]
In fact, I think it would be a good idea not allowing monitoring functions directly at all (a function depends on inputs by definition so it doesn't make much sense to store a function of a particular input with the function name itself, does it?)

So `monitor testFn(X)` or `monitor testFn` that currently work, would also trigger an error instructing the user to replace the monitor call to this instead:

`monitor monitorTestFnLabel = testFn(X)`

In this sense, we are invoking the storage of a variable (label name) which is computed as the function for the particular input, in this case X

**Comment by kupiqu** [comment]
Unfortunately, I realize that we use direct monitoring of functions all the time, so it would mean a drastic change for DynaSim :/

**Comment by kupiqu** [comment]
Fixed with pull request #644

**Comment by kupiqu** [comment]
it is possible now to monitor linkers directly #646
